evap/staff/importers.py

```diff
@@ -97,28 +98,99 @@ def store_in_database(self, vote_start_datetime, vote_end_date, semester):
+
+    def check_name_collision(self, semester):
+        name_collision = None
+        if Course.objects.filter(semester=semester, name_en=self.name_en).exists():
+            name_collision = "name_en"
+        if Course.objects.filter(semester=semester, name_de=self.name_de).exists():
+            name_collision = "name_de" if not name_collision else "both"
+        return name_collision
+
+    def find_and_set_existing_course(self, semester, importer):
+        courses = Course.objects.filter(
             semester=semester,
+            name_en=self.name_en,
+            name_de=self.name_de,
         )
-        course.save()
-        course.responsibles.set([responsible_dbobj])
-        course.degrees.set(self.degrees)
-        evaluation = Evaluation(
-            vote_start_datetime=vote_start_datetime,
-            vote_end_date=vote_end_date,
-            course=course,
-            wait_for_grade_upload_before_publishing=self.is_graded,
-        )
-        evaluation.save()
-        evaluation.contributions.create(
-            evaluation=evaluation,
-            contributor=responsible_dbobj,
-            role=Contribution.Role.EDITOR,
-            textanswer_visibility=Contribution.TextAnswerVisibility.GENERAL_TEXTANSWERS,
-        )
+        course = courses.first()
+        if course is None:
+            return
+
+        differing_attributes = []
+        if course.type != self.course_type:
+            differing_attributes.append("course type")
+        if course.responsibles.count() != 1 or course.responsibles.first().email != self.responsible_email:
+            differing_attributes.append("responsible person")
+        if not differing_attributes:
+            evaluations = Evaluation.objects.filter(course=course)
+            if len(evaluations) == 1:
+                if evaluations.first().wait_for_grade_upload_before_publishing == self.is_graded:
+                    self.existing_course = course
+                else:
+                    importer.errors[ImporterError.COURSE].append(
+                        _(
+                            "Course {} ({}) does already exist in this semester but the grading of the evaluation does not match."
+                        ).format(self.name_en, self.name_de)
+                    )
+            else:
+                importer.errors[ImporterError.COURSE].append(
+                    _(
+                        "Course {} ({}) does already exist in this semester and is identical but must have exactly one evaluation."
+                    ).format(self.name_en, self.name_de)
+                )
+        else:
+            importer.errors[ImporterError.COURSE].append(
+                _("Course {} ({}) does already exist in this semester but the {} {} not match.").format(
+                    self.name_en,
+                    self.name_de,
+                    " and ".join(differing_attributes),
+                    ngettext("does", "do", len(differing_attributes)),
```

Comment:
This does not work for translations. In german, the words "does" and "do" would disappear -- we wouldn't have: "aber die Attribute X und Y tun nicht übereinstimmen", instead the "match" would be the word we conjugate. You will have to put a bit more context into the translated phrase to make it work -- probably two copies of the phrase, one for singular and one for plural, is what we need.

Reply:
https://github.com/e-valuation/EvaP/pull/1683/commits/8b7fbc5ba9de0aa6e0d5efcd3f0e704acf339e0b should fix this problem, right?

Reply:
If I understand correctly, the "the" and " and " are not translated at all right now